src/frontend/src/lib/components/signer/SignerConsentMessage.svelte

```diff
@@ -0,0 +1,153 @@
+<script lang="ts">
+	import { Markdown } from '@dfinity/gix-components';
+	import type {
+		icrc21_consent_info,
+		ConsentMessageApproval,
+		Rejection
+	} from '@dfinity/oisy-wallet-signer';
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { getContext } from 'svelte';
+	import { fade } from 'svelte/transition';
+	import SignerLoading from '$lib/components/signer/SignerLoading.svelte';
+	import SignerOrigin from '$lib/components/signer/SignerOrigin.svelte';
+	import ButtonGroup from '$lib/components/ui/ButtonGroup.svelte';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { SIGNER_CONTEXT_KEY, type SignerContext } from '$lib/stores/signer.store';
+	import { toastsError } from '$lib/stores/toasts.store';
+
+	const {
+		consentMessagePrompt: { payload, reset: resetPrompt },
+		callCanisterPrompt: { reset: resetCallCanisterPrompt }
+	} = getContext<SignerContext>(SIGNER_CONTEXT_KEY);
+
+	let approve: ConsentMessageApproval | undefined;
+	$: approve = $payload?.status === 'result' ? $payload?.approve : undefined;
+
+	let reject: Rejection | undefined;
+	$: reject = $payload?.status === 'result' ? $payload?.reject : undefined;
+
+	let consentInfo: icrc21_consent_info | undefined;
+	$: consentInfo = $payload?.status === 'result' ? $payload?.consentInfo : undefined;
```

Comment:
can the three declarations above be merged?

`{ approve, reject, consentInfo } = $payload?.status...`

Reply:
Sure. Done.

```
let approve: ConsentMessageApproval | undefined;
	let reject: Rejection | undefined;
	let consentInfo: icrc21_consent_info | undefined;

	$: ({ approve, reject, consentInfo } =
		nonNullish($payload) && $payload.status === 'result'
			? $payload
			: { approve: undefined, reject: undefined, consentInfo: undefined });
```